From bd9ab360a81bc07e1139be0f9b42aaf17946b584 Mon Sep 17 00:00:00 2001
From: Anca-Jeanina Floarea <anca.floarea@freescale.com>
Date: Wed, 8 Aug 2012 18:18:43 +0000
Subject: [PATCH 267/518] dpa_stats: Added support for checking return code

The return code for functions put_req and set_cnt_ipsec_cb was
not verified against an error.

Signed-off-by: Anca Jeanina FLOAREA <anca.floarea@freescale.com>
[Grabbed from the branch, LINUX_IR5.2.0, of
https://git.freescale.com/git-private/cgit.cgi/ppc/alu-b4860/linux.git.]
Signed-off-by: Tiejun Chen <tiejun.chen@windriver.com>
---
 drivers/staging/fsl_dpa_offload/dpa_stats.c |    6 +++++-
 1 files changed, 5 insertions(+), 1 deletions(-)

diff --git a/drivers/staging/fsl_dpa_offload/dpa_stats.c b/drivers/staging/fsl_dpa_offload/dpa_stats.c
index 02df48b..05dfb50 100644
--- a/drivers/staging/fsl_dpa_offload/dpa_stats.c
+++ b/drivers/staging/fsl_dpa_offload/dpa_stats.c
@@ -2035,6 +2035,10 @@ void async_req_work_func(struct work_struct *work)
 		}
 
 		err = put_req(dpa_stats, req_cb);
+		if (err < 0) {
+			pr_err("Failed to release request control block\n");
+			req_cb->bytes_num = -ENOEXEC;
+		}
 
 		/* Notify the application */
 		req_cb->request_done(0, req_cb->config.storage_area_offset,
@@ -2233,7 +2237,7 @@ int dpa_stats_create_counter(int dpa_stats_id,
 		cnt_cb->f_get_cnt_stats = get_cnt_ccnode_stats;
 		break;
 	case DPA_STATS_CNT_IPSEC:
-		set_cnt_ipsec_cb(cnt_cb, params);
+		err = set_cnt_ipsec_cb(cnt_cb, params);
 		if (err != 0) {
 			pr_err("Failed to create IPSec counter");
 			goto create_counter_err;
-- 
1.7.5.4

